### PR TITLE
Use glBlitFramebuffer when rendering

### PR DESCRIPTION
**Pull request by robert-ancell**
This is much faster than using a shader which is not required currently (we are not doing any transformations).

### CODE REVIEW

**Comment by robert-ancell** [comment]
Done as part of diagnosing https://github.com/flutter/flutter/issues/148653. It hasn't resolved the issue, but has made things simpler (and faster) than before.

**Review by jonahwilliams — COMMENTED** [review]
seems strictly better, but also blitFramebuffer is an OpenGL 3 proc. I dont know what the minimum gl version the linux embedder requires, but something to keep in mind.

File: shell/platform/linux/fl_renderer.cc
Comment (loic-sharma) [inline shell/platform/linux/fl_renderer.cc:227]: FYI, you'll probably want to disable the stencil test here. See this comment by @bdero: https://github.com/flutter/engine/pull/49726#discussion_r1451038236

Corresponding Windows logic:

https://github.com/flutter/engine/blob/a338661d909c51dc15aedcba0978a80ec3a3c234/shell/platform/windows/compositor_opengl.cc#L142-L160

**Comment by robert-ancell** [comment]
> seems strictly better, but also blitFramebuffer is an OpenGL 3 proc. I dont know what the minimum gl version the linux embedder requires, but something to keep in mind.

It sounds like practically this will exist, but we should provide a fallback to cover all cases.

**Comment by robert-ancell** [comment]
Moving this to draft status as I need to think about an appropriate fallback behaviour.

**Comment by jonahwilliams** [comment]
If you are on OpenGL 2 or less then I think the current shader based solution is the best you can do. Still, that is an ancient gl version, and I don't know if we actually support it

**Comment by robert-ancell** [comment]
Talking to the OpenGL experts here it seems GLES2 is the minimum we can expect, but practically all systems will have GL3. So we just need to provide some form of fallback for a rare case - i.e. it probably doesn't have to be performant.

File: shell/platform/linux/fl_renderer.cc
Comment (loic-sharma) [inline shell/platform/linux/fl_renderer.cc:165]: Is this right?

```suggestion
                      geometry.y + geometry.height, GL_COLOR_BUFFER_BIT,
```

**Review by loic-sharma — APPROVED** [review]
LGTM but my understanding of OpenGL is shaky. Please also get approval from someone with a strong understanding like @jonahwilliams.

Also, don't forget to get a test exemption before landing this.

File: shell/platform/linux/fl_renderer.cc
Comment (robert-ancell) [reply shell/platform/linux/fl_renderer.cc]: Oh, whoops. Yes.

File: shell/platform/linux/fl_renderer.cc
Comment (cbracken) [inline shell/platform/linux/fl_renderer.cc:113]: nit: here and below: I assume that `GLint` is defined as `int` consistently in every implementation but I thought we used `GLint` against GL APIs elsewhere.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Comment by chinmaygarde** [comment]
Doesn't this need a fallback for ES2?

**Comment by robert-ancell** [comment]
> Doesn't this need a fallback for ES2?

If glBlitFramebuffer it reverts to the current texture rendering - is there something special in ES2 that would stop that working?

File: shell/platform/linux/fl_renderer.cc
Comment (robert-ancell) [reply shell/platform/linux/fl_renderer.cc:113]: Agree, we should use the GL types here. Note this is in the existing code, not the change added in this PR.

**Comment by robert-ancell** [comment]
Requesting a test exemption as this render path is not explicitly checked currently.

**Comment by robert-ancell** [comment]
Added a test, so this no longer requires the exemption.

**Pull request merged**
